crates/ruma-client-api/src/r0.rs

```diff
@@ -12,6 +12,7 @@ pub mod device;
 pub mod directory;
 pub mod filter;
 pub mod keys;
+pub mod knock;
```

Comment:
Yes, this should be feature-gated, since it's not part of a released version of the spec yet. Maybe we should consider having separate feature flags for things that are part of the unstable spec vs. things that are only in MSCs in the future.

Reply:
I've used `unstable-pre-spec` for now.